### PR TITLE
init: use core22 base for `snapcraft init`

### PR DESCRIPTION
**Pull request by mr-cal**
- [X] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [X] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [X] Have you successfully run `make lint`?
- [X] Have you successfully run `pytest tests/unit`?

-----

:heavy_check_mark: Change `snapcraft init`'s default base from `core20` to `core22`
:heavy_check_mark: Move command from `snapcraft_legacy/` to `snapcraft/`
:heavy_check_mark: Add a spread test

#### Source
- https://bugs.launchpad.net/snapcraft/+bug/1990550

(CRAFT-1670)

### CODE REVIEW

File: snapcraft.spec [hunk 1]
```diff
@@ -18,7 +18,7 @@ data += copy_metadata("lazr.uri")
 data += copy_metadata("wadllib")
 
 a = Analysis(
-    ["snapcraft_legacy\\cli\\__main__.py"],
+    ["snapcraft\\__main__.py"],
```
Comment (mr-cal) [inline snapcraft.spec:21]: I believe this is the correct thing to do.

This is the entrypoint for the Windows snapcraft executable.  We've migrated many commands from `snapcraft_legacy` to `snapcraft`, but `init` is the only command that is [tested](https://github.com/snapcore/snapcraft/blob/ddd328f07f1aa64c63907b9ffbd5eafa11818dde/appveyor.yml#L64) on Windows by Appveyor.

File: tests/spread/general/initialize/task.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+summary: Run snapcraft init
+
+restore: |
+  rm -f ./*.snap
+
+execute: |
+  # 'snapcraft init' should create a usable snapcraft.yaml file
+  snapcraft init
+  snapcraft
```
Comment (tigarmo) [inline tests/spread/general/initialize/task.yaml:9]: for the sake of completeness, should we check that the `snapcraft.yaml` file has `base: core22` in it?

File: snapcraft/commands/initialize.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    parts:
+      my-part:
+        # See 'snapcraft plugins'
+        plugin: nil
+    """
+)
+
+
+class InitializeCommand(BaseCommand, abc.ABC):
```
Comment (tigarmo) [inline snapcraft/commands/initialize.py:52]: I see that you changed `InitCommand` to `InitializeCommand` to distinguish from Python's `__init__()`, right? Can you talk a bit more about that reasoning? Did you see an instance here the two could be confused somehow?
The reason I ask is that I think that consistency with rock/charmcraft is a compelling reason to keep the command as `InitCommand`; so unless you feel strongly about this change I'd vote for changing back.

File: snapcraft.spec (hunk 1 above)
Comment (tigarmo) [reply snapcraft.spec:21]: looks good to me, but I definitely don't have the whole context of Windows snapcraft
@sergiusens do you see any harm in this?

File: snapcraft/commands/initialize.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
```
Comment (tigarmo) [inline snapcraft/commands/initialize.py:1]: Other than init -> initialize and the updated template, are there are changes in this file that I should look at? Or is the rest of it just from the move from snapcraft_legacy?

File: snapcraft/commands/initialize.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        except errors.ProjectMissing:
+            emit.progress("Could not find an existing 'snapcraft.yaml'.")
+
+        snapcraft_yaml_path = Path("snap/snapcraft.yaml")
+
+        emit.progress(f"Creating {str(snapcraft_yaml_path)!r}.")
+
+        snapcraft_yaml_path.parent.mkdir(exist_ok=True)
+
+        with open(snapcraft_yaml_path, mode="w", encoding="utf-8") as file:
```
Comment (tigarmo) [inline snapcraft/commands/initialize.py:81]:
```suggestion
        snapcraft_yaml_path.write_text(_TEMPLATE_YAML)
```

File: tests/unit/commands/test_initialize.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
```
Comment (tigarmo) [inline tests/unit/commands/test_initialize.py:1]: same question here: are there specific changes that I should check here?

File: snapcraft/commands/initialize.py (hunk 3 above)
Comment (mr-cal) [reply snapcraft/commands/initialize.py:52]: That's a good point.  I don't feel strongly about it and didn't think about the consistency with the other craft applications.  I'll keep it is `InitCommand`.

File: snapcraft/commands/initialize.py (hunk 4 above)
Comment (mr-cal) [reply snapcraft/commands/initialize.py]: Functionally, it's identical.  There are two minor changes under the hood:
1. `os.path` -> `pathlib.Path`
2. Using [`get_snap_project()`](https://github.com/snapcore/snapcraft/blob/ddd328f07f1aa64c63907b9ffbd5eafa11818dde/snapcraft/parts/lifecycle.py#L81) to locate an existing snapcraft.yaml

File: tests/unit/commands/test_initialize.py (hunk 6 above)
Comment (mr-cal) [reply tests/unit/commands/test_initialize.py]: It's a rewrite from `unit.TestCase` to `pytest`, but there's not one thing that stands out but everything stands out :laughing:

File: tests/spread/general/initialize/task.yaml (hunk 2 above)
Comment (mr-cal) [reply tests/spread/general/initialize/task.yaml]: Done!

File: snapcraft/commands/initialize.py (hunk 5 above)
Comment (mr-cal) [reply snapcraft/commands/initialize.py]: Done!

File: snapcraft.spec (hunk 1 above)
Comment (sergiusens) [reply snapcraft.spec:21]: this is correct

**Review by sergiusens — APPROVED** [review]

**Pull request merged**